marin/post_training/environments/swe_bench_env.py

```diff
@@ -0,0 +1,285 @@
+import json
+import os
+import random
+import subprocess
+import uuid
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dataclasses import dataclass
+from typing import Any
+
+import datasets
+import jax
+import numpy as np
+from post_training.inference import batch_inference
+from swebench.inference.make_datasets.utils import extract_diff
+from tqdm.auto import tqdm
+
+from .marin_env import EnvStep, MarinEnv
+
+
+@dataclass
```

Comment:
nit: `@dataclass(frozen=True)` since I think it remains static.

Reply:
Thanks for the suggestion. Updated.